x-pack/plugins/cross_cluster_replication/public/app/app.js

```diff
@@ -19,47 +36,186 @@ import {
+            fetchPermissionError: error.data,
+          });
+        }
+
+        // This error isn't an HTTP error, so let the fatal error screen tell the user something
+        // unexpected happened.
+        fatalError(error, i18n.translate('xpack.crossClusterReplication.app.checkPermissionsFatalErrorTitle', {
+          defaultMessage: 'Cross Cluster Replication app',
+        }));
+      }
+    }
+
+    registerRouter() {
+      const { router } = this.context;
+      routing.reactRouter = router;
+    }
+
+    render() {
+      const {
+        isFetchingPermissions,
+        fetchPermissionError,
+        hasPermission,
+        missingPermissions,
+      } = this.state;
+
+      if (isFetchingPermissions) {
+        return (
+          <EuiPageContent horizontalPosition="center">
+            <EuiFlexGroup>
+              <EuiFlexItem>
+                <EuiLoadingSpinner size="xl"/>
+              </EuiFlexItem>
+
+              <EuiFlexItem>
+                <EuiTitle>
+                  <h2>
+                    <FormattedMessage
+                      id="xpack.crossClusterReplication.app.permissionCheckTitle"
+                      defaultMessage="Checking permissions..."
+                    />
+                  </h2>
+                </EuiTitle>
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </EuiPageContent>
+        );
+      }
+
+      if (fetchPermissionError) {
+        const { error: errorString, statusCode, message } = fetchPermissionError;
+
+        return (
+          <EuiCallOut
+            title={(
+              <FormattedMessage
+                id="xpack.crossClusterReplication.app.permissionCheckErrorTitle"
+                defaultMessage="Error checking permissions"
+              />
+            )}
+            color="danger"
+            iconType="alert"
+          >
+            {`${statusCode}: ${errorString}. ${message}`}
+          </EuiCallOut>
+        );
+      }
+
+      if (!hasPermission) {
+        return (
+          <EuiPageContent horizontalPosition="center">
+            <EuiEmptyPrompt
+              iconType="securityApp"
+              iconColor={null}
+              title={
+                <h2>
+                  <FormattedMessage
+                    id="xpack.crossClusterReplication.app.deniedPermissionTitle"
+                    defaultMessage="Permission denied"
+                  />
+                </h2>}
+              body={
+                <p>
+                  <FormattedMessage
+                    id="xpack.crossClusterReplication.app.deniedPermissionDescription"
+                    defaultMessage="You do not have required permissions ({permissions}) for Cross Cluster Replication."
+                    values={{ permissions: missingPermissions.join(', ') }}
+                  />
+                </p>}
+            />
+          </EuiPageContent>
+        );
+      }
+
+      if (!isAvailable() || !isActive()) {
```

Comment:
I can't seem to hit this path. when I have a basic license, I get:

![image](https://user-images.githubusercontent.com/1965714/51768932-85979d80-2096-11e9-949f-bb0de0175613.png)

maybe this condition needs to be moved above the other blocks (line 177)?

Reply:
Great catch! Fixed.